### PR TITLE
Change description format to reflect its action v2

### PR DESCRIPTION
**Pull request by peterfication**
See https://github.com/johmsalas/text-case.nvim/pull/47 and https://github.com/johmsalas/text-case.nvim/issues/49

### CODE REVIEW

File: lua/textcase/shared/utils.lua [hunk 1]
```diff
@@ -104,10 +104,11 @@ local callableTable = {
   end
 }
 
-function utils.create_wrapped_method(desc, method)
+function utils.create_wrapped_method(method_name, method, desc)
```
Comment (peterfication) [inline lua/textcase/shared/utils.lua:107]: This is a breaking change, but the utils are kind of internal I guess.

A search in Github reveals no other hits than this repo or forks of it: https://github.com/search?q=create_wrapped_method+language%3ALua&type=code&l=Lua

But this won't reveal usages in private repos of course.

File: lua/textcase/plugin/plugin.lua [hunk 2]
```diff
@@ -23,6 +24,8 @@ function M.register_methods(method_table, opts)
   -- TODO: validate method_table
   M.state.methods_by_desc[method_table.desc] = method_table
```
Comment (peterfication) [inline lua/textcase/plugin/plugin.lua:25]: The `methods_by_desc` is not used anymore after this change. So it could be removed.

**Comment by peterfication** [comment]
This is not working yet.

**Comment by peterfication** [comment]
This change is more involved than originally planned. If it was that complex, I probably wouldn't have made the original PR. 

So the question is whether it's worth it or not.

**Comment by johmsalas** [comment]
In my personal workflow, I take advantage of the fuzzy search provided by telescope. Therefore I barely read the texts on the screen and their text form is not an indicator of the item to be selected.

Do you have some thoughts or any personal workflow @peterfication?

**Comment by peterfication** [comment]
I use telescope as well and sometimes the keybindings with which-key. The user experience felt nicer to me if the description provided a better view of what the action will do. Especially if I didn't use it in a long time and forgot the name for pascal case for example.

But if I used it more regularly, I wouldn't look at it too much anymore I guess.

**Comment by peterfication** [comment]
I would not be sad if this PR is not being merged in.

**Comment by chrisgrieser** [comment]
I simply define the keymaps myself via lua require, and then I can also see desc of my own. That way, I got nice which-key descriptions

**Comment by peterfication** [comment]
So in the end, @chrisgrieser you would agree, that it would be nicer to have self explanatory descriptions? I know, you have it on your own via your config. However, I assume a lot of people are using this plugin just as is. And they would profit from it.

**Comment by chrisgrieser** [comment]
yeah sure they would be better. But I dunno, it feels like unnecessary to unnecessarily break the config of an unknown percentage of users other that?

**Comment by peterfication** [comment]
I already commented in the issue you opened that I agree and that it should definitely not break the configuration of users. Hence, I created this PR which does not break the config as far as I can tell.

However, it misses test coverage for that which is an issue of its own.

**Comment by johmsalas** [comment]
@peterfication what is missing on your PR for being ready for review? I do like you decoupled presentational aspect from behavior so I'm inclined towards merging it

**Comment by peterfication** [comment]
It's basically ready, just that there are no unit tests. I use it locally also with custom keybindings and it works for me. I hope I didn't miss a manual test.

I would definitely appreciate if @chrisgrieser could try out this version and see whether it works for him.

**Comment by chrisgrieser** [comment]
works fine on my end!

**Comment by johmsalas** [comment]
Couldn't test it myself. I'm afk this week. But it is start of day for me, it will be easy to revert it if I get "paged"

Thanks @peterfication @chrisgrieser. I built this plugin for my specific requirement. Let me know if you have interest on having more permissions on the repo

**Comment by peterfication** [comment]
Sure, would be interesting to support you in maintaining this plugin 👍 My next step would probably be looking into the unit tests to get them running.

**Pull request merged**